README.md

```diff
@@ -101,47 +101,91 @@ libp2p becomes very simple and basically acts as a glue for every module that co
 //   transport: websockets + tcp
-//   stream-muxing: SPDY
+//   stream-muxing: spdy & mplex
 //   crypto-channel: secio
 //   discovery: multicast-dns
 
 const libp2p = require('libp2p')
 const TCP = require('libp2p-tcp')
 const WS = require('libp2p-websockets')
 const spdy = require('libp2p-spdy')
+const mplex = require('libp2p-mplex')
 const secio = require('libp2p-secio')
 const MulticastDNS = require('libp2p-mdns')
 const DHT = require('libp2p-kad-dht')
+const defaultsDeep = require('lodash.defaultsdeep')
 
 class Node extends libp2p {
-  constructor (peerInfo, peerBook, options) {
-    options = options || {}
-
-    const modules = {
-      transport: [
-        new TCP(),
-        new WS()
-      ],
-      connection: {
-        muxer: [
-          spdy
+  constructor (_peerInfo, _peerBook, _options) {
+    const options = {
+      peerInfo: _peerInfo             // The Identity of your Peer
+      peerBook: _peerBook,            // Where peers get tracked, if undefined libp2p will create one instance
+
+      // The libp2p modules for this libp2p bundle
+      modules: {
+        transport: [
+          TCP,
+          WS
         ],
-        crypto: [
-          secio
+        streamMuxer: [
+          SPDY
+        ],
+        connEncryption: [
+          SECIO
         ]
+        peerDiscovery: [
+          MulticastDNS
+        ],
+        peerRouting: {},              // Currently both peerRouting and contentRouting are patched through the DHT,
+        contentRouting: {}            // this will change once we factor that into two modules, for now do the following line:
+        dht: DHT                      // DHT enables PeerRouting, ContentRouting and DHT itself components
+      },
+
+      // libp2p config options (typically found on a config.json)
+      config: {                       // The config object is the part of the config that can go into a file, config.json.
+        addresses: {                  // Multiaddrs
+          listen: []                  // To start transport listener
+          announce: []                // To announce/share with the network (i.e DNS addr)
+          noAnnounce: []              // To keep private
+        },
+        peerDiscovery: {
+          mdns: {                     // mdns options
+            interval: 1000            // ms
+            enabled: true
+          },
+          webrtcStar: {               // webrtc-star options
+            interval: 1000            // ms
+            enabled: false
+          }
+          // .. other discovery module options.
+        },
+        peerRouting: {},
+        contentRouting: {},
+        protocolMuxing: {             // Protocol muxing sequence. libp2p uses sane defaults (below). You can override these with:
+          transport: {
+            secio: {
+              spdy: '*',              // '*' means, from this point forward, accept any protocol
+              mplex: '*'
+            }
+          }
+        }
```

Comment:
This feature will have to be implemented at the Switch Level

Reply:
Before merging, I'll remove these options from the README and continue working on this feature design at https://github.com/libp2p/js-libp2p/issues/201